pkg/download/buffer.go

```diff
@@ -105,25 +105,22 @@ func (m *BufferMode) Fetch(ctx context.Context, url string) (io.Reader, int64, e
 	fileSize := firstReqResult.fileSize
 	trueURL := firstReqResult.trueURL
 
-	if fileSize <= m.minChunkSize() {
+	if fileSize <= m.chunkSize() {
 		// we only need a single chunk: just download it and finish
 		return br, fileSize, nil
 	}
 
-	remainingBytes := fileSize - m.minChunkSize()
-	numChunks := int(remainingBytes / m.minChunkSize())
+	remainingBytes := fileSize - m.chunkSize()
+	numChunks := int(remainingBytes / m.chunkSize())
```

Comment:
this isn't right; it's the right calculation for minChunkSize (where we want to guarantee each chunk is _at least_ minChunkSize big, but it's the wrong calculation for chunkSize (where chunkSize is an _upper bound_ on the size of a chunk so we don't grow our buffers too big)

Reply:
Took me a while to work out why our tests weren't failing but I got a failing test case in 3254d15ab8af7d32787d04c4dbfc9eaef18329b5. I'll push a fix later.